cql3/expr/expression.cc

```diff
@@ -903,6 +903,29 @@ expression replace_column_def(const expression& expr, const column_definition* n
         }, expr);
 }
 
+// Replaces all occurences of token(p1, p2) on the left hand side with the given colum.
+// For example this changes token(p1, p2) < token(1, 2) to my_column_name < token(1, 2).
+expression replace_token(const expression& expr, const column_definition* new_cdef) {
```

Comment:
This function always copies the entire expression - even in the common case that there is no token() in it at all. I wonder if there is no way to modify the expression in-place. Maybe this is not critical performance-wise, though? I see there's other similar code already existing in expression.cc. @dekimir please review this patch (at least), you're the expert in this code.

Reply:
I think it's only ever called when preparing a statement.

Reply:
I generally frown upon modifying AST in place.  Too many painful memories brought by that practice.  If this proves to be a performance bottleneck, we can revisit.  Absent that, it's fine to copy.